src/controllers/midi/midicontroller.h

```diff
@@ -102,6 +105,11 @@ class MidiController : public Controller {
     MidiControllerPreset m_preset;
     SoftTakeoverCtrl m_st;
     QList<QPair<MidiInputMapping, unsigned char> > m_fourteen_bit_queued_mappings;
+    MixxxClock m_mixxxClock;
```

Comment:
The clocksource class takes a clock as an argument by pointer, so some object needs to own the original clock.  Since the MixxxClock is extremely lightweight, it's not a problem to just have it in every midicontroller instance.  I could move the clock up the stack farther, but that doesn't seem to help.  Alternatively, I could move the mock/clock code into time.h and have an alternative constructor for clocksource that doesn't take an argument and instead asks time.h for a static clock source object